src/transformStateWithClones.js

```
@@ -1,13 +1,70 @@
 'use strict';
 
 /**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
+ * @param {Object} state Estado inicial
+ * @param {Object[]} actions Lista de ações
+ * @return {Object[]} Lista de estados clonados após cada ação
  */
-function transformStateWithClones(state, actions) {
-  // write code here
+const transformStateWithClones = (state, actions) => {
+  const stateCopy = { ...state };
+  const states = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        addProperties(stateCopy, action.extraData);
+        break;
+      }
+
+      case 'removeProperties': {
+        removeProperties(stateCopy, action.keysToRemove);
+        break;
+      }
+
+      case 'clear': {
+        clearState(stateCopy);
+        break;
+      }
+
+      default: {
+        break;
+      }
```

This violates checklist item #3: 'switch/case should always have default case for error handling.' An empty `default` block silently ignores unknown action types, which can lead to unexpected behavior. It's better to explicitly handle this case, for example, by throwing an error.